src/data/mv_polynomial/basic.lean

```diff
@@ -102,6 +102,9 @@ instance [monoid R] [comm_semiring S₁] [distrib_mul_action R S₁] :
   distrib_mul_action R (mv_polynomial σ S₁) :=
 add_monoid_algebra.distrib_mul_action
 
+instance [comm_semiring R] [smul_zero_class S₁ R] : smul_zero_class S₁ (mv_polynomial σ R) :=
```

Comment:
The instance above uses the variables in the other order,
```suggestion
instance [comm_semiring S₁] [smul_zero_class R S₁] : smul_zero_class R (mv_polynomial σ S₁) :=
```

Reply:
I feel a little confused in the variable naming in this file.
The doc comment said, "`R : Type*` `[comm_semiring R]` (the coefficients)". But the actual usage is sometimes  `mv_polynomial σ R` with `[a-type-about-smul S₁ R]` (such as in `coeff_smul`),  and sometimes with opposite meanings of `R` and `S₁` (`mv_polynomial σ S₁` with `[a-type-about-smul R S₁]`).
https://github.com/leanprover-community/mathlib/blob/178a32653e369dce2da68dc6b2694e385d484ef1/src/data/mv_polynomial/basic.lean#L466-L468
https://github.com/leanprover-community/mathlib/blob/178a32653e369dce2da68dc6b2694e385d484ef1/src/data/mv_polynomial/basic.lean#L105-L106

Reply:
I'm afraid the answer is that:
* Over the whole file it's just a mess, as you rightly observe
* Fixing that mess in mathlib3 is just going to be annoying to forward-port
* In the meantime, local consistency is better than doing what the docstring thinks we should have done!